biogeochem/EDPhysiologyMod.F90

```diff
@@ -1330,6 +1336,238 @@ subroutine phenology_leafonoff(currentSite)
+
+    class(prt_vartypes), pointer :: prt
+
+    ! !LOCAL VARIABLES:                                                                                            
+    type(ed_patch_type) , pointer :: currentPatch     
+    type(ed_cohort_type), pointer :: currentCohort
+
+    real(r8) ::  spread        ! dummy value of canopy spread to estimate c_area
+    real(r8) ::  leaf_c        ! leaf carbon estimated to generate target tlai
+    real(r8) ::  sumarea
+    real(r8) :: check_treelai
+    integer ::   fates_pft     ! fates pft numer for weighting loop
+    integer  ::   hlm_pft      ! host land model pft number for weighting loop.
+    integer ::   s             ! site index
+
+
+   ! To Do in this routine.
+   ! Get access to HLM input varialbes. 
+   ! Weight them by PFT
+   ! Loop around patches, and for each single cohort in each patch 
+   ! call assign_cohort_SP_properties to determine cohort height, dbh, 'n',  area, leafc from drivers.
+
+   currentSite%sp_tlai(:) = 0._r8
+   currentSite%sp_tsai(:) = 0._r8
+   currentSite%sp_htop(:) = 0._r8
+
+   ! WEIGHTING OF FATES PFTs on to HLM_PFTs
+   ! 1. Add up the area associated with each FATES PFT 
+   ! where pft_areafrac is the area of land in each HLM PFT and (from surface dataset)
+   ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
+
+   ! 2. weight each fates PFT target for lai, sai and htop by the area of the 
+   ! contrbuting HLM PFTs.
+
+   currentPatch => currentSite%oldest_patch
+   do while (associated(currentPatch))
+
+     fates_pft = currentPatch%nocomp_pft_label
+     if(fates_pft.ne.0)then 
+
+       sumarea = 0.0_r8
+       do hlm_pft = 1,size( EDPftvarcon_inst%hlm_pft_map,2)
+
+         if(bc_in%pft_areafrac(hlm_pft) * EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft).gt.0.0_r8)then
+            sumarea = sumarea + bc_in%pft_areafrac(hlm_pft)*EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft)
+            !leaf area index
+            currentSite%sp_tlai(fates_pft) = currentSite%sp_tlai(fates_pft) + &
+                    bc_in%hlm_sp_tlai(hlm_pft) * bc_in%pft_areafrac(hlm_pft) &
+                    * EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft) 
+           !stem area index
+           currentSite%sp_tsai(fates_pft) = currentSite%sp_tsai(fates_pft) + &
+                 bc_in%hlm_sp_tsai(hlm_pft) *	bc_in%pft_areafrac(hlm_pft) &      		       
+               * EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft)
+           ! canopy height
+           currentSite%sp_htop(fates_pft) = currentSite%sp_htop(fates_pft) + &
+                 bc_in%hlm_sp_htop(hlm_pft) * bc_in%pft_areafrac(hlm_pft) &
+               * EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft)
+         end if ! there is some area in this patch
+       end do !hlm_pft
+
+       ! weight for total area in each patch/fates_pft
+       if(currentPatch%area.gt.0.0_r8)then 
+         currentSite%sp_tlai(fates_pft) = currentSite%sp_tlai(fates_pft) &
+             /(currentPatch%area/area)
```

Comment:
It's because the areas by which the lai values etc are multiplied in the previous section don't necessarily add up to one.  

tlai  = sum( sp_tlai(hlmpft) * pftareafrac(hlmpft) * hlm_pft_map(fates_pft,hlm_pft)
          
say that a given fates PFT's patch is made up 50% of the grid cell, which was itself made up from 30% of one pft and 20% from another. The input lais are both 10, that would give
10*0.3*1. + 10*0.2+1.  = 2+3 = 5

the right answer here is 10, (for the weighted average lai) and so we need ti divide by 5000/10000.= 0.5 (or x2) 

To check this I just did a simulation where I commented out these lines and the LAI was too small compared with the CLM5-SP run LAI (more so in areas with mixed pfts like savanna and W Europe). 

Does that make sense.  I added another comment to help out here.

Reply:
Thanks for the clarification. From your example, I now understand that "currentPatch%area/area" actually equal to "sumarea". I suppose "currentPatch%area" has been defined according to "bc_in%pft_areafrac(hlm_pft)" and "EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft)" in SP mode, right?  :)